fixed subprocess call to work on all systems

dalanicolai opened the pull request:
Changed the use of 'top' to the use of 'ps', because the top command did not work for on systems with custom top configuration file.
See [this](https://stackoverflow.com/questions/23989095/how-do-you-extract-only-the-cpu-usage-and-the-process-names-columns-from-top-com) thread for more info.

Also replaced the Popen (incl.PIPE) for a simple check_output command.

Sorry forgot to adapt the column numbers, see the comment below. However I don't know how to adapt them quickly so I hope you can very quickly adapt them in line number 119-121 while reviewing the merge...

dalanicolai committed:
fixed subprocess call to work on all systems

Changed the use of 'top' to the use of 'ps', because the top command did not work for on systems with custom top configuration file.
See [this](https://stackoverflow.com/questions/23989095/how-do-you-extract-only-the-cpu-usage-and-the-process-names-columns-from-top-com) thread for more info.

Also replaced the Popen (incl.PIPE) for a simple check_output command.

dalanicolai reviewed (commented):
I forgot to adapt the column numbers in the for loop. Of course they should changed to the following
        pid = col[0]
        cpu = atof(col[1])
        cmd = ' '.join(col[2:])

gornostal commented:
@dalanicolai sorry, I had to revert it because I get this error:

```
Traceback (most recent call last):
  File "/home/agornostal/projects/ulauncher/ulauncher/api/client/Client.py", line 54, in on_message
    self.extension.trigger_event(event)
  File "/home/agornostal/projects/ulauncher/ulauncher/api/client/Extension.py", line 52, in trigger_event
    action = listener.on_event(event, self)
  File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 44, in on_event                                                                                       
    return RenderResultListAction(list(islice(self.generate_results(event), 15)))
  File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 47, in generate_results                                                                               
    for (pid, cpu, cmd) in get_process_list():
  File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 118, in get_process_list                                                                              
    cpu = atof(col[8])
  File "/usr/lib/python3.7/locale.py", line 326, in atof
    return func(delocalize(string))
ValueError: could not convert string to float: '--no-startvm-errormsgbox'
```

dalanicolai commented:
Hi Aleksandr,

Yes sorry. It was a small fix. So I fixed it directly on github (instead of
forking). I added in a comment to the merge that I forgot to adapt the
column numbers (see here). And also I gave the right numbers there. If I
knew how, then of course I would have fixed the column numbers very quickly
myself. Anyway, the extension was not working on my system, and will not on
systems with a custom top configuration. I will issue a "re-revert"
request. And attach the comment again...

(this was the comment)
I forgot to adapt the column numbers in the for loop. Of course they should
changed to the following
pid = col[0]
cpu = atof(col[1])
cmd = ' '.join(col[2:])

Hope it is fine...



On Sun, 27 Oct 2019 at 18:27, Aleksandr Gornostal <notifications@github.com>
wrote:

> @dalanicolai <https://github.com/dalanicolai> sorry, I had to revert it
> because I get this error:
>
> Traceback (most recent call last):
>   File "/home/agornostal/projects/ulauncher/ulauncher/api/client/Client.py", line 54, in on_message
>     self.extension.trigger_event(event)
>   File "/home/agornostal/projects/ulauncher/ulauncher/api/client/Extension.py", line 52, in trigger_event
>     action = listener.on_event(event, self)
>   File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 44, in on_event
>     return RenderResultListAction(list(islice(self.generate_results(event), 15)))
>   File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 47, in generate_results
>     for (pid, cpu, cmd) in get_process_list():
>   File "/home/agornostal/.local/share/ulauncher/extensions/com.github.ulauncher.ulauncher-kill/main.py", line 118, in get_process_list
>     cpu = atof(col[8])
>   File "/usr/lib/python3.7/locale.py", line 326, in atof
>     return func(delocalize(string))
> ValueError: could not convert string to float: '--no-startvm-errormsgbox'
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ulauncher/ulauncher-kill/pull/5?email_source=notifications&email_token=AEMTOX4L4MMBDQTDEC7KPJ3QQW6PHA5CNFSM4JD7MBO2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECLDS4Q#issuecomment-546716018>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEMTOX4O4Z6VSV5CE2GLT2TQQW6PHANCNFSM4JD7MBOQ>
> .
>

gornostal commented:
I guess I misunderstood your comment. I thought you've fixed it.

> I will issue a "re-revert" request. And attach the comment again...

Cool! Thanks for contributing to this extension :+1:

gornostal commented:
> And attach the comment again...

Wait. What do you mean by that? :)

dalanicolai commented:
Ah okay great!  Only see your answers now. Thanks!
